### PR TITLE
Updated storage layer diagram

### PR DESCRIPTION
**Pull request by holisticode**
This PR is just to update the storage layer image

### CODE REVIEW

**Comment by holisticode** [comment]
@zelig I believe I would like to keep mockstore in the diagram mostly for us developers; how mockstore plugs in never hasn't been obvious to me, and this might be helpful to others in the team (nevertheless, being high level, it may also add much either).

LRU cache visually documents we are using Least Recently Used cache strategy, and LDBStore explains what DB implementation we are using, which is kind of interesting information nowhere else documented.

Streamer just shows how the network and storage layer interface. I am happy to remove the Streamer part, although we *do* explain syncing so it may be useful.

In short: I am defending the diagram as-is, I think it has its merits in documenting little-documented parts of the code. Nevertheless I have no strong feelings in removing the mentioned parts if you insist, specifically also if others chime in voicing their opinions.

**Comment by justelad** [comment]
@holisticode can you reopen against `master`? @zelig can we merge as is?
